pkg/hubble/relay/observer/server.go

```diff
@@ -231,6 +232,103 @@ func (s *Server) GetNodes(ctx context.Context, req *observerpb.GetNodesRequest)
 	return &observerpb.GetNodesResponse{Nodes: nodes}, nil
 }
 
+// GetNamespaces implements observerpb.ObserverClient.GetNamespaces.
+func (s *Server) GetNamespaces(ctx context.Context, req *observerpb.GetNamespacesRequest) (*observerpb.GetNamespacesResponse, error) {
```

Comment:
We should re-use the `namespaceManager` here in order to reduce the logic duplication. Something like:

1. create a `NewNamespaceManager()`
2. for each peer, add each ns to the manager (`(*namespaceManager).AddNamespace()` will handle the dedup and is already concurrency safe)
3. return the manager's `GetNamespaces` result (will handle the sorting).

What do you think?

Reply:
Hmm, I'm not sure I completely agree. The namespaceManager is keeping state over time, whereas the relay isn't, it's constructing everything within the RPC. We don't need expiration logic, etc, from the namespaceManager. I agree though that this could be refactored to share some more code between the two servers. The way they handle concurrency is a bit different too. The namespaceManager uses mutex because there's many readers, one writer, but in relay, we have many writers, zero readers (we read _after_ the results get aggregated). I don't think the same approach makes sense for both.

I think what would make more sense is to introduce a `NamespaceSet` or something which could reduce duplication some of the logic for managing the map, and then turning it into a deduplicated list.

Reply:
Okay, with some refactoring I think it can be reused, and the RWMutex is semantically the same as the channel based approach, and the overhead of adding the time tracking for each namespace is probably negligible.

Reply:
feels much cleaner thank you. I don't think there is anything wrong to re-use the `namespaceManager` here even though we do not care about time tracking.